### PR TITLE
Add support for shared resources

### LINKED ISSUE #74: Shared resource directory is ignored
The resource directory in the "shared" folder is not added to project `resourceDirectories`. This is a problem especially for `CrossType.Pure` as all resource directories for this layout are in the "hidden" `.<platform>` folder.

Ideally, resource directories should be treated in the same way as source directories are.

build.sbt:
```scala
import sbtcrossproject.{crossProject, CrossType}

lazy val foo =
  crossProject(JSPlatform, JVMPlatform)
    .crossType(CrossType.Pure)

lazy val `foo-jvm` = foo.jvm
lazy val `foo-js`  = foo.js
```
SBT shell:
```
sbt:tmp> show fooJVM/sourceDirectories
[info] * /private/tmp/foo/.jvm/src/main/scala-2.12
[info] * /private/tmp/foo/.jvm/src/main/scala
[info] * /private/tmp/foo/.jvm/src/main/java
[info] * /private/tmp/foo/src/main/scala-2.12
[info] * /private/tmp/foo/src/main/scala
[info] * /private/tmp/foo/.jvm/target/scala-2.12/src_managed/main
sbt:tmp> show fooJVM/resourceDirectories
[info] * /private/tmp/foo/.jvm/src/main/resources
[info] * /private/tmp/foo/.jvm/target/scala-2.12/resource_managed/main
```

### PR DESCRIPTION
**Pull request by armanbilge**
Fixes #74.

### CODE REVIEW

**Comment by armanbilge** [comment]
The test is a bit simpler than I had hoped, but I think it's sufficient.

File: sbt-crossproject/src/main/scala/sbtcrossproject/CrossType.scala
Comment (sjrd) [inline sbt-crossproject/src/main/scala/sbtcrossproject/CrossType.scala:45]: Hum, we cannot add an abstract method. That is not backward binary compatible, as it will break existing subclasses in the wild that do not implement that method. Normally the CI would catch that, but we had not set up MiMa yet. I submitted a PR #134 to set it up.

Instead, this method should be concrete, with a default implementation (probably `= None`).

Comment (sjrd) [reply sbt-crossproject/src/main/scala/sbtcrossproject/CrossType.scala:45]: Also, since this adds a public method, it will require a minor version bump. So you should first include a commit (in this PR) that bumps the next version to 1.1.0-SNAPSHOT (in `project/Extra.scala`), and then a second commit with these changes.

**Review by sjrd — CHANGES_REQUESTED** [review]

File: sbt-crossproject/src/main/scala/sbtcrossproject/CrossType.scala
Comment (armanbilge) [reply sbt-crossproject/src/main/scala/sbtcrossproject/CrossType.scala]: Oops, right. Fixed. Do you want me to rebase against #134 / master once its merged?

Comment (sjrd) [reply sbt-crossproject/src/main/scala/sbtcrossproject/CrossType.scala]: Yes, ideally we should wait for #134 to be merged and then rebase this one on top afterwards.

**Comment by sjrd** [comment]
#134 was merged. Would you mind rebasing on top of the latest master?

**Comment by djspiewak** [comment]
Once this is merged, could we get a release which supports it? The epoch directories have been needed for a long, long time.

**Comment by armanbilge** [comment]
@sjrd rebased. And +1 on the release, pretty please :)

**Review by sjrd — APPROVED** [review]

**Pull request merged**
